src/components/Grid/Grid.tsx

```diff
@@ -0,0 +1,725 @@
+      if (e.buttons === NO_BUTTONS_PRESSED || dragState.current === false) {
+        return;
+      }
+
+      containerRef.current?.setPointerCapture(dragState.current);
+    }, []);
+
+    const onPointerUp: PointerEventHandler<HTMLDivElement> = useCallback(
+      e => {
+        if (containerRef.current) {
+          containerRef.current.releasePointerCapture(e.pointerId);
+          if (!contextMenuOpen) {
+            containerRef.current.focus();
+          }
+        }
+
+        dragState.current = false;
+        onFocusChange(focus.row, focus.column);
+      },
+      [contextMenuOpen, focus.column, focus.row, onFocusChange]
+    );
+
+    const onMouseMove: MouseEventHandler<HTMLDivElement> = useCallback(
+      async e => {
+        e.preventDefault();
+        e.stopPropagation();
+        if (
+          dragState.current === false ||
+          e.buttons === NO_BUTTONS_PRESSED ||
+          !containerRef.current ||
+          !outerRef.current
+        ) {
+          return;
+        }
+
+        containerRef.current.setPointerCapture(dragState.current);
+        let verticalSign = 0,
+          horizontalSign = 0,
+          x = e.clientX,
+          y = e.clientY;
+        if (
+          elementBorderRef.current.left >= e.clientX ||
+          e.clientX >=
+            elementBorderRef.current.right - elementBorderRef.current.scrollBarWidth
+        ) {
+          const scrollHorizontalDirection =
+            e.clientX <= elementBorderRef.current.left ? "left" : "right";
+          if (
+            (scrollHorizontalDirection === "left" && e.movementX < 0) ||
+            (scrollHorizontalDirection === "right" && e.movementX > 0)
+          ) {
+            horizontalSign = scrollHorizontalDirection === "left" ? -1 : 1;
+          }
+          x =
+            scrollHorizontalDirection === "left"
+              ? elementBorderRef.current.left + rowNumberWidth + 10
+              : elementBorderRef.current.right -
+                elementBorderRef.current.scrollBarWidth -
+                10;
+        }
+
+        if (
+          elementBorderRef.current.top >= e.clientY ||
+          e.clientY >=
+            elementBorderRef.current.bottom - elementBorderRef.current.scrollBarHeight
+        ) {
+          const scrollVerticalDirection =
+            e.clientY <= elementBorderRef.current.top ? "top" : "bottom";
+          if (
+            (scrollVerticalDirection === "top" && e.movementY < 0) ||
+            (scrollVerticalDirection === "bottom" && e.movementY > 0)
+          ) {
+            verticalSign = scrollVerticalDirection === "top" ? -1 : 1;
+          }
+          y =
+            scrollVerticalDirection === "top"
+              ? elementBorderRef.current.top + headerHeight + 10
+              : elementBorderRef.current.bottom -
+                elementBorderRef.current.scrollBarHeight -
+                10;
+        }
+
+        await new Promise(requestAnimationFrame);
```

Comment:
explain why we are doing this

Reply:
This is for the scroll to be processed and element to be visible
After the scroll immediately we won't be able to fetch the item

Reply:
yeah, but lets put a comment on the code.